solenspipe/bias.py

```diff
@@ -488,3 +625,388 @@ def RDN0_analytic(shape,wcs,theory,fwhm,noise_t,noise_p,powdict,estimator,XY,UV,
                                  Aalpha=AXY,Abeta=AUV,xmask=xmask,ymask=ymask,kmask=kmask,
                                  field_names_alpha=None,field_names_beta=None,skip_filter_field_names=False,
                                  split_estimator=split_estimator)
+
+def mcrdn0_s4(icov, get_kmap, power,phifunc, nsims, qfunc1,get_kmap1=None,get_kmap2=None,get_kmap3=None, qfunc2=None, Xdat=None,Xdat1=None,Xdat2=None,Xdat3=None, use_mpi=True, 
```

Comment:
get_kmap1 etc. should be normal rather than optional arguments since the function won't work when they have their default value of `None`.